File: src/trackers/es/hdcity_li.ts
import {WebChecker} from "../../checker/webChecker";
import {CheckerResult} from "../../checker/checkerResult";

export default class Hdcity_li extends WebChecker {
    public get url() { return "https://hdcity.li"; }
    public get name() { return "hdcity.li"; }
    public get language() { return "es"; }
    public get categories(): Array<String> { return ["movies"]; }
    public get description() { return ""; }
    public get registryUrl() { return "https://hdcity.li/index.php?page=account"; }

    async isOpen(): Promise<CheckerResult> {
        let $ = await WebChecker.getQueryUrl(this.registryUrl);
        let exists = $(":contains('but registrations are closed')").length > 0;

        return {
            online: true,
            regClosed: !exists,
            regOpened: false
        };
    }
}

File: checker/trackers/hdcityli.go
package trackers

import (
	"github.com/tokkenno/open-torrent-tracker-bot/checker/check"
	"github.com/tokkenno/open-torrent-tracker-bot/models"
	"strings"
)

type HDCityLi struct {
}

func (tracker *HDCityLi) GetId() string {
	return "hdcity_li"
}

func (tracker *HDCityLi) GetName() string {
	return "HDCity"
}

func (tracker *HDCityLi) GetDescription() string {
	return ""
}

func (tracker *HDCityLi) GetLanguages() []string {
	return []string{"es"}
}

func (tracker *HDCityLi) GetCategories() []string {
	return []string{"movies"}
}

func (tracker *HDCityLi) GetUrl() string {
	return "https://hdcity.li"
}

func (tracker *HDCityLi) GetRegistryUrl() string {
	return "https://hdcity.li/index.php?page=account"
}

func (tracker *HDCityLi) Check() *check.Result {
	html, err := check.GetWebCode(tracker.GetRegistryUrl())

	if err != nil {
		return &check.Result{
			IsOnline: false,
		}
	} else {
		var status models.RegistrationStatus

		if strings.Contains(html, "registrations are closed") {
			status = models.RegistrationClose
		} else {
			status = models.RegistrationMaybe
		}

		return &check.Result{
			IsOnline: true,
			Status:   status,
		}
	}
}
